### ISSUE TITLE
uSync 13 content with form (macro) produces error

### ISSUE THREAD

**Issue by Nuklon** [Jumoo/uSync.Complete.Issues#216]
**Describe the bug**
Error while processing Push - (export/Dependencies/0/)

```csharp
Object reference not set to an instance of an object.
   at Umbraco.Cms.Infrastructure.Persistence.Querying.ExpressionVisitorBase.VisitMethodCall(MethodCallExpression m)
   at Umbraco.Cms.Infrastructure.Persistence.Querying.ExpressionVisitorBase.Visit(Expression expression)
   at Umbraco.Cms.Infrastructure.Persistence.Querying.ExpressionVisitorBase.VisitLambda(LambdaExpression lambda)
   at Umbraco.Cms.Infrastructure.Persistence.Querying.ExpressionVisitorBase.Visit(Expression expression)
   at Umbraco.Cms.Infrastructure.Persistence.Querying.Query`1.Where(Expression`1 predicate)
   at Umbraco.Cms.Infrastructure.Persistence.Repositories.Implement.MacroRepository.PerformGetByAlias(String alias)
   at Umbraco.Cms.Core.Cache.DefaultRepositoryCachePolicy`2.Get(TId id, Func`2 performGet, Func`2 performGetAll)
   at Umbraco.Cms.Infrastructure.Persistence.Repositories.Implement.MacroRepository.GetByAlias(String alias)
   at Umbraco.Cms.Core.Services.MacroService.GetByAlias(String alias)
   at uSync.Core.Mapping.MacroMapper.GetDependencies(Object value, String editorAlias, DependencyFlags flags)
   at uSync.Core.Mapping.RTEMapper.GetDependencies(Object value, String editorAlias, DependencyFlags flags)
   at uSync.Core.Mapping.SyncValueMapperCollectionExtensions.GetDependencies(SyncValueMapperCollection mapperCollection, Object value, String editorAlias, DependencyFlags flags)
   at uSync.Core.Mapping.SyncNestedValueMapperBase.GetPropertyDependencies(JObject value, IContentType docType, DependencyFlags flags)
   at uSync.Core.Mapping.BlockListMapper.GetDependencies(Object value, String editorAlias, DependencyFlags flags)
   at uSync.Expansions.Core.Dependency.ContentBaseChecker.GetPropertyDependencies(IContentBase item, DependencyFlags flags)
   at uSync.Expansions.Core.Dependency.ContentChecker.GetDependencies(IContent item, DependencyFlags flags)
   at uSync.BackOffice.SyncHandlers.SyncHandlerRoot`2.GetDependencies(TObject item, DependencyFlags flags)
   at uSync.BackOffice.SyncHandlers.SyncHandlerRoot`2.GetDependencies(Guid key, DependencyFlags flags)
   at uSync.Expansions.Core.Cache.SyncCacheService.GetInternalDependency(ISyncHandler handler, Udi udi, DependencyFlags flags)
   at uSync.Expansions.Core.Services.SyncDependencyService.GetHandlerDependencies(ISyncHandler handler, String handlerSet, Udi udi, DependencyFlags flags)
   at uSync.Expansions.Core.Services.SyncDependencyService.GetItemDependency(SyncItem item, String handlerSet, IList`1 existing)
   at uSync.Expansions.Core.Services.SyncDependencyService.GetDependencies(SyncPackRequest request)
   at uSync.Publisher.Services.SyncPublishService.GetDependencies(SyncPublishRequest request)
   at uSync.Publisher.Publishers.SyncRealtimePublisher.PushCalculate(PublisherActionRequest request)
   at uSync.Publisher.Services.SyncPublisherActionService.PerformAction(PublisherActionRequest request, ISyncPublisher publisher, IUser user)
```

**Version (please complete the following information):**
 - Umbraco Version: 13 RC3
 - uSync.Complete Version 13 RC3

**To Reproduce**
1. Add a form.
2. Add some content and add macro to use said form.
3. Push content to other server/
4. See error.

**Comment by Nuklon** [comment]
To clarify, I'm also using the uSync Forms RC.

**Comment by KevinJump** [comment]
yeah - when i could actually get a Macro to stay in the editor (Umbraco keeps wiping them out 🤷 ) I have got a fix for this (in uSync, - the new block lists in RTE format wasn't being parsed for dependencies).

_Nightly Build: https://dev.azure.com/jumoo/Public/_artifacts/feed/nightly/NuGet/uSync/overview/13.0.0-rc2.20231201.6_

**Comment by Nuklon** [comment]
Thanks, can confirm it now runs OK. I also had it with another macro, which didn't work anymore after saving the document.

**Comment by KevinJump** [comment]
Closed as fixed in uSync 13.0 release.
